### DISCUSSION TITLE
Support for FontAwesome Icons on MARP preview

### DISCUSSION THREAD

**Discussion by Saptarshi-max** [marp-team/marp#83] [Q&A] [answered]
This has been a highly productive VSCode extension for me, and perfect for developing presentations a Big thanks for building this amazing platform to the authors.

Recently for a presentation, I needed some Icons in my slides. The![FontAwesome](https://fontawesome.com/) is a big repository of icons few of which I will be using for my slides. To import Fontawesome icons on a generic Markdown file, The following HTML code can be copy-pasted

```html
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet"/>
``` 
and wherever I will like to put in my icon of choice I need to use another HTML snippet.

```html
<h4><i class="fa fa-flag" aria-hidden="true"></i> One Font, 675 Icons, and add other texts here</h4>
```
Where fa-flag is the name of the icon I wanted to import. But in VSCode, even with the HTML code execution enabled the MARP preview is unable to display these icons. However, these icons are perfectly imported on the final 'pdf' output of the 'Export slide deck'. Is there any way I can see the FontAwesome Icons on the MARP slide preview beforehand ? ... Visualizing the Icons only on the final slide pdf output and not on the preview itself is very time consuming

**Comment by yhatt** [comment, marked as answer]
VS Code preview looks like sanitizing `<link>` tag due to some security reason.

Instead, you can import the external CSS by [`@import` rule](https://developer.mozilla.org/en-US/docs/Web/CSS/@import) within inline style ([`<style>` tag for tweaking theme CSS](https://marpit.marp.app/theme-css?id=tweak-style-through-markdown)).

```html
<style>
@import 'https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css';
</style>
```

![](https://user-images.githubusercontent.com/3993388/113325878-ef392380-9353-11eb-9fec-41c96355725e.png)

**Reply by Saptarshi-max** [reply]
Awesome, It works now. 
Thank you so much for your quick response.

**Reply by majkinetor** [reply]
Doesn't work when slide is exported to HTML. The one without @import works.

**Comment by KhaaL** [comment]
https://github.com/marp-team/marp/discussions/83



For everyone else who might've tried this solution without success: make sure to close the preview pane and re-enable the "Toggle Marp Feature for current markdown" from the Marp icon in VScode.

Also, make sure to enable _Markdown › Marp: Enable HTML_

Took me days of troubleshooting 🤦🏽‍


@yhatt, is it possible to use the new FA fonts in Marp? I tried with this CDN, but I only got blank blocks in the renderer.

https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@6.1.2/css/fontawesome.min.css


***

**Reply by yhatt** [reply]
@KhaaL CSS file you've presented looks like no references to any Web font resources (`@font-face`). You can use https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@6.1.2/css/all.min.css instead.

**Reply by KhaaL** [reply]
Thank you for the prompt response (and all the awesome work in this project) 🙂

**Reply by josethz00** [reply]
thank you! this does the trick

**Comment by kazumatu981** [comment]
[FYI]: I was released markdow-it plugin for marp `markdown-it-fontawesome`.

[@kazumatu981/markdown-it-fontawesome - npm](https://www.npmjs.com/package/@kazumatu981/markdown-it-fontawesome)

To use this, you can write easier decks inclueing Fontawesome icons.
Please Enjoy.

**Reply by Saptarshi-max** [reply]
Thank you for sharing!
